src/promptflow/tests/executor/e2etests/test_activate.py

```diff
@@ -65,52 +77,78 @@ def test_all_nodes_bypassed(self, dev_connections):
 
     def test_aggregate_bypassed_nodes(self):
         flow_folder = "conditional_flow_with_aggregate_bypassed"
-        batch_engine = BatchEngine(get_yaml_file(flow_folder), get_flow_folder(flow_folder), connections={})
+        mem_run_storage = MemoryRunStorage()
+        batch_engine = BatchEngine(
+            get_yaml_file(flow_folder), get_flow_folder(flow_folder), connections={}, storage=mem_run_storage
+        )
         input_dirs = {"data": get_flow_inputs_file(flow_folder, file_name="inputs.json")}
         inputs_mapping = {"case": "${data.case}", "value": "${data.value}"}
         output_dir = Path(mkdtemp())
         batch_results = batch_engine.run(input_dirs, inputs_mapping, output_dir)
 
         expected_result = get_flow_expected_result(flow_folder)
         expected_status_summary = get_flow_expected_status_summary(flow_folder)
-        self.assert_activate_bulk_run_result(batch_results, expected_result, expected_status_summary)
+        self.assert_activate_bulk_run_result(
+            output_dir, mem_run_storage, batch_results, expected_result, expected_status_summary
+        )
 
         # Validate the aggregate result
-        assert batch_results.aggr_results.node_run_infos["aggregation_double"].output == 3
-        assert batch_results.aggr_results.node_run_infos["aggregation_square"].output == 12.5
-
-    def assert_activate_bulk_run_result(self, result: BatchResult, expected_result, expected_status_summary):
+        for node_run_info in mem_run_storage._node_runs.values():
+            if node_run_info.node == "aggregation_double":
+                assert node_run_info.status == Status.Completed
+                assert node_run_info.output == 3
+            elif node_run_info.node == "aggregation_square":
+                assert node_run_info.status == Status.Completed
+                assert node_run_info.output == 12.5
+
+    def assert_activate_bulk_run_result(
+        self,
+        output_dir: Path,
+        mem_run_storage: MemoryRunStorage,
+        batch_result: BatchResult,
+        expected_result,
+        expected_status_summary,
+    ):
         # Validate the flow outputs
-        for i, output in enumerate(result.outputs):
+        outputs = load_jsonl(output_dir / OUTPUT_FILE_NAME)
+        for i, output in enumerate(outputs):
             expected_outputs = expected_result[i]["expected_outputs"].copy()
             expected_outputs.update({"line_number": i})
             assert output == expected_outputs
 
         # Validate the flow line results
-        for i, line_result in enumerate(result.line_results):
-            self.assert_activate_flow_run_result(line_result, expected_result[i])
+        flow_runs = {k: v for k, v in sorted(mem_run_storage._flow_runs.items(), key=lambda item: item[1].index)}
```

Comment:
Will this be better for understanding?
```suggestion
        flow_runs = {k: v for k, v in sorted(mem_run_storage._flow_runs.items(), key=lambda _, v: v.index)}
```

Reply:
Updated.

Reply:
But it can't work. I reverted it.